plugins/chanbackup.c

```diff
@@ -0,0 +1,328 @@
+
+	/* The header part */
+	if (crypto_secretstream_xchacha20poly1305_init_pull(&crypto_state, final,
+							    (&secret)->data) != 0){
+		plugin_log(p, LOG_DBG, "SCB file is corrupted!");
+		return 0;
+	}
+
+	if (crypto_secretstream_xchacha20poly1305_pull(&crypto_state, ans,
+						       NULL, 0,
+						       final + crypto_secretstream_xchacha20poly1305_HEADERBYTES,
+						       st.st_size - crypto_secretstream_xchacha20poly1305_HEADERBYTES,
+						       NULL, 0) != 0){
+		plugin_log(p, LOG_DBG, "SCB file is corrupted!");
+		return 0;
+	}
+	
+	close(fd);
+
+	return ans;
+}
+
+static struct command_result *after_recover_rpc(struct command *cmd,
+					 const char *buf,
+					 const jsmntok_t *params,
+					 void *cb_arg UNUSED)
+{
+
+	int i = 0;
+	const jsmntok_t *t;
+	struct json_stream *response;
+
+	response = jsonrpc_stream_success(cmd);
+
+	json_for_each_obj(i, t, params)
+		json_add_tok(response, json_strdup(tmpctx, buf, t), t+1, buf);
+
+	return command_finished(cmd, response);
+}
+
+/* Recovers the channels by making RPC to `recoverchannel` */
+static struct command_result *recover(struct command *cmd,
+					      const char *buf,
+					      const jsmntok_t *params)
+{
+	struct out_req *req;
+
+	if (!param(cmd, buf, params, NULL))
+		return command_param_failed();
+	
+	const char *res = (char *)decrypt_scb(cmd->plugin);
+
+	req = jsonrpc_request_start(cmd->plugin, cmd, "recoverchannel",
+				    after_recover_rpc,
+				    &forward_error, NULL);
+	
+	const jsmntok_t *restok;
+	restok = json_parse_simple(tmpctx, res, strlen(res));
+
+	json_add_tok(req->js, "scb", restok, res);
+
+	return send_outreq(cmd->plugin, req);
+}
+
+static void update_scb(struct plugin *p, const char *scb_buf)
+{
+
+	/* If the temp file existed before, remove it */
+	unlink_noerr("scb.tmp");
+	
+	int fd = open("scb.tmp", O_CREAT|O_EXCL|O_WRONLY, 0400);
+
+	plugin_log(p, LOG_DBG, "Updating the SCB file...");
+
+	write_hsm(p, fd, scb_buf);
+
+	/*~ fsync (mostly!) ensures that the file has reached the disk. */
+	if (fsync(fd) != 0) {
+		unlink_noerr("scb.tmp");
+	}
+
+	/*~ This should never fail if fsync succeeded.  But paranoia good, and
+	 * bugs exist. */
+	if (close(fd) != 0) {
+		unlink_noerr("scb.tmp");
+	}
+	/*~ We actually need to sync the *directory itself* to make sure the
+	 * file exists!  You're only allowed to open directories read-only in
+	 * modern Unix though. */
+	fd = open(".", O_RDONLY);
+	if (fd < 0) {
+		plugin_log(p, LOG_DBG, "Opening: %s", strerror(errno));
+	}
+	if (fsync(fd) != 0) {
+		unlink_noerr("scb.tmp");
+	}
+	close(fd);
+
+	rename("scb.tmp", "scb");
+}
+
+
+static struct command_result *after_staticbackup(struct command *cmd,
+					 const char *buf,
+					 const jsmntok_t *params,
+					 void *cb_arg UNUSED)
+{
+
+	const char *scb_buf =  json_strdup(tmpctx, buf, params);
+	const jsmntok_t *scb_arr = json_parse_simple(tmpctx, scb_buf, strlen(scb_buf)),
+					*scbs = json_get_member(scb_buf, scb_arr, "scb");
```

Comment:
You can just do `const jmsntok_t *scbs = json_get_member(params, "scb")`.  You are converting the JSON to a string and reparsing it, which is unnecessary!